test/jdk/java/awt/Dialog/EnabledResetTest.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 1999, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 4232374
+ * @summary Tests that dismissing a modal dialog does not enable
+ *          disabled components
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual EnabledResetTest
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.Dialog;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.Window;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class EnabledResetTest {
+    public static void main(String[] args) throws Exception {
+        String INSTRUCTIONS = """
+                1. Press "Create Child" twice to create three windows
+                   Verify that the parent windows are disabled
+                2. Press "Create Modal Dialog"
+                   Verify that the parent windows are disabled
+                3. Press "enable"
+                   Verify that no windows accept mouse events
+                4. Press "ok"
+                   Verify that the first window is still disabled
+                   If all the verifications are done, then test is
+                   PASSED, else test fails.
+                                              """;
```

Comment:
Closing quotes placement here and at other places
```suggestion
                   If all the verifications are done, then test is
                   PASSED, else test fails.
                   """;
```

Reply:
Updated.